src/lxc/tools/lxc_unshare.c

```diff
@@ -164,7 +164,7 @@ static bool lookup_user(const char *oparg, uid_t *uid)
 
 	if (sscanf(oparg, "%u", uid) < 1) {
 		/* not a uid -- perhaps a username */
-		if (sscanf(oparg, "%s", name) < 1) {
+		if (strlen(name) >= PATH_MAX || sscanf(oparg, "%s", name) < 1) {
```

Comment:
Hmm, actually, that looks wrong. At the point you're doing the strlen check, name has never been assigned, it will only get assigned through the following sscanf call.

Coverity agrees:
```
Hi,

Please find the latest report on new defect(s) introduced to lxc/lxc found with Coverity Scan.

1 new defect(s) introduced to lxc/lxc found with Coverity Scan.


New defect(s) Reported-by: Coverity Scan
Showing 1 of 1 defect(s)


** CID 1593169:  Uninitialized variables  (UNINIT)
/src/lxc/tools/lxc_unshare.c: 167 in lookup_user()


________________________________________________________________________________________________________
*** CID 1593169:  Uninitialized variables  (UNINIT)
/src/lxc/tools/lxc_unshare.c: 167 in lookup_user()
161             buf = malloc(bufsize);
162             if (!buf)
163                     return false;
164     
165             if (sscanf(oparg, "%u", uid) < 1) {
166                     /* not a uid -- perhaps a username */
>>>     CID 1593169:  Uninitialized variables  (UNINIT)
>>>     Using uninitialized value "*name" when calling "strlen".
167                     if (strlen(name) >= PATH_MAX || sscanf(oparg, "%s", name) < 1) {
168                             free(buf);
169                             return false;
170                     }
171     
172                     ret = getpwnam_r(name, &pwent, buf, bufsize, &pwentp);
```

Reply:
lol, yes.